MoreLinq.Test/FlattenTest.cs

```diff
@@ -289,5 +294,164 @@ public void FlattenEvaluatesInnerSequencesLazily()
             Assert.Throws<TestException>(() =>
                 source.Flatten().ElementAt(11));
         }
+
+        // Flatten(this IEnumerable source, Func<object, IEnumerable> selector)
+
+        [Test]
+        public void FlattenSelectorIsLazy()
+        {
+            new BreakingSequence<int>().Flatten(BreakingFunc.Of<object, IEnumerable>());
+        }
+
+        [Test]
+        public void FlattenSelector()
+        {
+            var source = new[]
+            {
+                new Series
+                {
+                    Name = "series1",
+                    Attributes = new[]
+                    {
+                        new Attribute { Values = new[] { 1, 2 } },
+                        new Attribute { Values = new[] { 3, 4 } },
+                    }
+                },
+                new Series
+                {
+                    Name = "series2",
+                    Attributes = new[]
+                    {
+                        new Attribute { Values = new[] { 5, 6 } },
+                    }
+                }
+            };
+
+            var result = source.Flatten(obj =>
+            {
+                switch (obj)
+                {
+                    case string str:
+                        return null;
+                    case IEnumerable inner:
+                        return inner;
+                    case Series s:
+                        return new object[] { s.Name, s.Attributes };
+                    case Attribute a:
+                        return a.Values;
+                    default:
+                        return null;
+                }
+            });
+
+            var expectations = new object[] { "series1", 1, 2, 3, 4, "series2", 5, 6 };
+
+            Assert.That(result, Is.EquivalentTo(expectations));
```

Comment:
`EquivalentTo` is not an _exact_ test of expectations. Per [docs](https://github.com/nunit/docs/wiki/CollectionEquivalentConstraint):

> **CollectionEquivalentConstraint** tests that two `IEnumerables` are equivalent - that they contain the same items, in any order.

```suggestion
            Assert.That(result, Is.EqualTo(expectations));
```